pgmpy/models/MarkovModel.py

```diff
@@ -676,3 +676,43 @@ def get_partition_function(self):
             raise ValueError('Factor for all the random variables not defined.')
 
         return np.sum(factor.values)
+
+    def copy(self):
+        """
+        Returns a copy of this Markov Model.
+ 
+        Returns
+        -------
+        MarkovModel: Copy of this Markov model.
+ 
+        Examples
+        -------
+        >>> from pgmpy.factors import Factor
+        >>> from pgmpy.models import MarkovModel
+        >>> G = MarkovModel()
+        >>> G.add_nodes_from([('a', 'b'), ('b', 'c')])
+        >>> G.add_edge(('a', 'b'), ('b', 'c'))
+        >>> G_copy = G.copy()
+        >>> G_copy.edges()
+        [(('a', 'b'), ('b', 'c'))]
+        >>> G_copy.nodes()
+        [('a', 'b'), ('b', 'c')]
+        >>> factor = Factor([('a', 'b')], cardinality=[3],
+        ...                 values=np.random.rand(3))
+        >>> G.add_factors(factor)
+        >>> G.get_factors()
+        [<Factor representing phi(('a', 'b'):3) at 0x...>]
+        >>> G_copy.get_factors()
+        []
+        """
+        clone_graph = MarkovModel()
+
+        clone_graph.add_nodes_from(self.nodes())
+        clone_graph.add_edges_from(self.edges())
```

Comment:
@liquidmetal You can remove this line and change line 708 to `clone_graph = MarkovModel(self.edges())`